src/riscv/lib/src/pvm/linux/signals.rs

```diff
@@ -282,19 +300,9 @@ impl<MC: MemoryConfig, M: ManagerReadWrite> MachineCoreState<MC, M> {
         let signal_index: SignalIndex = signal.into();
         let mask = self.signal_actions.read_mask(signal_index);
 
-        // TODO RV-756 Add support for changing signal disposition.
-        //
-        // For signals with a TERM or CORE disposition, we still want to change the program counter
-        // to 0 after the handler.
-        let pc = match signal {
-            Signal::Sigill
-            | Signal::Sigabrt
-            | Signal::Sigiot
-            | Signal::Sigbus
-            | Signal::Sigsegv
-            | Signal::Sigpipe
-            | Signal::Sigterm
-            | Signal::Sigsys => 0,
+        // For signals with a TERM or CORE disposition, we want to change the program counter to 0.
+        let pc = match Disposition::default(signal) {
+            Disposition::Term | Disposition::Core => 0,
             _ => self.hart.pc.read(),
         };
```

Comment:
Does this mean you return to PC 0 regardless of the current disposition of `signal`?